src/modules/asset-metrics/asset-metrics.service.ts

```diff
@@ -0,0 +1,106 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { AssetFundamentalsEntity, AssetMetricsDto, AssetMetricsEntity } from 'lib-typeorm';
+
+
+@Injectable()
+export class AssetMetricsService {
+  constructor(
+    @InjectRepository(AssetFundamentalsEntity)
+    private readonly assetFundamentalsRepository: Repository<AssetFundamentalsEntity>,
+    @InjectRepository(AssetMetricsEntity)
+    private readonly assetMetricsRepository: Repository<AssetMetricsEntity>,
+  ) {}
+
+  //METRICS SERVICES 
+
+  async createMetric (assetMetricsDto: AssetMetricsDto): Promise<AssetMetricsEntity> {
+    const existingMetric = await this.assetMetricsRepository.findOne({ where: { metric: assetMetricsDto.metric } });
+
+    if (existingMetric) {
+      throw new Error('Metric already exists');
+    }
+
+  
+    const assetMetric = this.assetMetricsRepository.create(assetMetricsDto)
+
+    return this.assetMetricsRepository.save(assetMetric);
+  }
+
+  async findAllMetrics(query: any): Promise<AssetMetricsEntity[]> {
+    const queryBuilder = this.assetMetricsRepository.createQueryBuilder('asset_metrics');
+    
+    if (query.metric) {
+      queryBuilder.andWhere('asset_metrics.metric = :metric', { metric : `%${query.metric}%` });
+    }
+
+    if (query.name) {
+      queryBuilder.andWhere('asset_metrics.name LIKE :name', { name : `%${query.name}%` });
+    }
+
+    if (query.category) {
+      queryBuilder.andWhere('asset_metrics.category = :category', { category : `%${query.category}%` });
+    }
+
+    if (query.subCategory) {
+      queryBuilder.andWhere('asset_metrics.subCategory = :subCategory', { subCategory : `%${query.subcategory}%` });
+    }
+
+    if (query.dataType) {
+      queryBuilder.andWhere('asset_metrics.dataType = :dataType', { dataType : `%${query.dataType}%` });
+    }
+
```

Comment:
same remark as below. Why we are using querybuilder with so many if statements?

Reply:
Used an alternative method now. Now i am using an array that stores the conditions for the queries